mesonbuild/compilers/mixins/clike.py

```diff
@@ -990,9 +990,24 @@ def symbols_have_underscore_prefix(self, env: 'Environment') -> bool:
 
     def _get_patterns(self, env: 'Environment', prefixes: T.List[str], suffixes: T.List[str], shared: bool = False) -> T.List[str]:
         patterns: T.List[str] = []
-        for p in prefixes:
+        if env.machines[self.for_machine].is_os2():
+            # On OS/2, search order for shared libs is
+            #   1. libfoo_dll.a
+            #   2. foo_dll.a
+            #   3. libfoo.a
+            #   4. foo.a
+            #   5. foo.dll
+            # For static libs, `_s' is used instead of `_dll'.
             for s in suffixes:
-                patterns.append(p + '{}.' + s)
+                for p in prefixes:
+                    if s.startswith('_dll.') or s.startswith('_s.'):
+                        patterns.append(p + '{}' + s)
+                    else:
+                        patterns.append(p + '{}.' + s)
```

Comment:
This can be cleaned up a little and fold the cases together:
```python
for s in suffixes
    dot = '.' if env[machines][self.for_machine].is_os2() and s.startswith(('_dll.', '_s.')):
    for p in prefixes:
        patterns.append('p + '{}' + dot + s)
```

Reply:
The last ':' of if statement is necessary?

Reply:
And I had to ':' to the end of 'for' statement line.